### PR TITLE
Fixed bugs found by psalm

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

When running psalm for branch 4.4, we get around 1500 errors. I browsed though them all and here is a small PR with the actual bugs I think should be fixed. 

I'll make review comments about them below.

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Command/LintCommand.php [hunk 1]
```diff
@@ -226,7 +226,7 @@ private function displayJson(OutputInterface $output, array $filesInfo)
         return min($errors, 1);
     }
 
-    private function renderException(OutputInterface $output, string $template, Error $exception, string $file = null)
+    private function renderException(SymfonyStyle $output, string $template, Error $exception, string $file = null)
```
Comment (Nyholm) [inline src/Symfony/Bridge/Twig/Command/LintCommand.php:229]: This is a minor. We pass a `SymfonyStyle` and we use the `$object` as a `SymfonyStyle`.

File: src/Symfony/Bridge/Twig/Mime/WrappedTemplatedEmail.php [hunk 2]
```diff
@@ -182,7 +182,7 @@ public function getBcc(): array
      */
     public function setPriority(int $priority): self
     {
-        $this->message->setPriority($priority);
+        $this->message->priority($priority);
```
Comment (Nyholm) [inline src/Symfony/Bridge/Twig/Mime/WrappedTemplatedEmail.php:185]: There is no function called `setPriority`. However, the `Email::priority()` exists. 

Im not sure if I should rename `WrappedTemplatedEmail::setPriority()` to `priority()` too.

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 3]
```diff
@@ -1564,7 +1564,7 @@ public function getContent($asResource = false)
      */
     public function getETags()
     {
-        return preg_split('/\s*,\s*/', $this->headers->get('if_none_match'), null, \PREG_SPLIT_NO_EMPTY);
+        return preg_split('/\s*,\s*/', $this->headers->get('if_none_match'), -1, \PREG_SPLIT_NO_EMPTY);
```
Comment (Nyholm) [inline src/Symfony/Component/HttpFoundation/Request.php:1567]: [preg_split](https://www.php.net/manual/en/function.preg-split.php)'s third argument supports `null` in PHP 7. However, in PHP 8 it is required to be an int. 

From PHPStorm's stubs: 

> If specified, then only substrings up to <i>limit</i> are returned with the rest of the string being placed in the last substring. A <i>limit</i> of -1, 0 or <b>NULL</b> means "no limit" and, as is standard across PHP, you can use <b>NULL</b> to skip to the <i>flags</i> parameter.

From the docs (PHP 8 ready): 

> If specified, then only substrings up to limit are returned with the rest of the string being placed in the last substring. A limit of -1 or 0 means "no limit".

File: src/Symfony/Component/VarDumper/Command/ServerDumpCommand.php [hunk 4]
```diff
@@ -95,5 +95,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $this->server->listen(function (Data $data, array $context, int $clientId) use ($descriptor, $io) {
             $descriptor->describe($io, $data, $context, $clientId);
         });
+
+        return 0;
```
Comment (Nyholm) [inline src/Symfony/Component/VarDumper/Command/ServerDumpCommand.php:99]: Just a command that does not return an int.

File: src/Symfony/Component/VarExporter/Internal/Exporter.php [hunk 5]
```diff
@@ -82,7 +82,7 @@ public static function prepare($values, $objectsPool, &$refsPool, &$objectsCount
                 }
 
                 if (!\is_array($properties = $value->__serialize())) {
-                    throw new \Typerror($class.'::__serialize() must return an array');
+                    throw new \TypeError($class.'::__serialize() must return an array');
```
Comment (Nyholm) [inline src/Symfony/Component/VarExporter/Internal/Exporter.php:85]: Typo

File: src/Symfony/Contracts/Service/ServiceSubscriberTrait.php [hunk 6]
```diff
@@ -43,7 +43,7 @@ public static function getSubscribedServices(): array
             }
 
             if (self::class === $method->getDeclaringClass()->name && ($returnType = $method->getReturnType()) && !$returnType->isBuiltin()) {
-                $services[self::class.'::'.$method->name] = '?'.($returnType instanceof \ReflectionNamedType ? $returnType->getName() : $type);
+                $services[self::class.'::'.$method->name] = '?'.($returnType instanceof \ReflectionNamedType ? $returnType->getName() : $returnType);
```
Comment (Nyholm) [inline src/Symfony/Contracts/Service/ServiceSubscriberTrait.php:46]: The variable `$type` is never defined in this class. I assume we mean `$returnType`.

File: src/Symfony/Component/DependencyInjection/Definition.php [hunk 7]
```diff
@@ -102,7 +102,7 @@ public function setChanges(array $changes)
     /**
      * Sets a factory.
      *
-     * @param string|array|Reference $factory A PHP function, reference or an array containing a class/Reference and a method to call
+     * @param string|array|Reference|null $factory A PHP function, reference or an array containing a class/Reference and a method to call
```
Comment (Nyholm) [inline src/Symfony/Component/DependencyInjection/Definition.php:105]: This is also a minor. 

The `SessionPass` is passing "null" as parameter to this method to reset the factory.

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 8]
```diff
@@ -59,7 +59,7 @@ public function copy($originFile, $targetFile, $overwriteNewerFiles = false)
             }
 
             // Stream context created to allow files overwrite when using FTP stream wrapper - disabled by default
-            if (false === $target = @fopen($targetFile, 'w', null, stream_context_create(['ftp' => ['overwrite' => true]]))) {
+            if (false === $target = @fopen($targetFile, 'w', false, stream_context_create(['ftp' => ['overwrite' => true]]))) {
```
Comment (Nyholm) [inline src/Symfony/Component/Filesystem/Filesystem.php:62]: Third parameter to [fopen](https://www.php.net/manual/en/function.fopen.php) bust be a boolean. Default is `false´.

From the docs: 

> The optional third use_include_path parameter can be set to '1' or true if you want to search for the file in the include_path, too.


Please very this extra carefully.

File: src/Symfony/Bridge/Twig/Mime/WrappedTemplatedEmail.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Bridge/Twig/Mime/WrappedTemplatedEmail.php:185]: the class is internal, no need to care about renaming, it won't improve anything

**Review by nicolas-grekas — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by jderusse — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @Nyholm.

**Pull request merged**